app/controllers/posts_controller.rb

```diff
@@ -30,6 +30,25 @@ def all
     @posts = Post.all.order("created_at DESC")
   end
 
+  # def destroy
+  #   return @error = true if @post.user_id != session[:user_id]
+  #   @post = Post.find(params[:id])
+  #   Post.where(user_id: session[:user_id]).delete(params[:id])
+  #   redirect_to user_posts_url(@user)
+  # end
+
+  def destroy
+    @user = User.find(session[:user_id])
```

Comment:
I understand your frustration about this section (as discussed) but I am also at a loss of how to flash to the user a message if they are not allowed.
What you have written is how I would have done it (there is routing and so logic sits on the controller).
The only other thing I thought to keep the controller skinny is to have some simple logic on the view that would only give the option to delete the post if it was the user.
Something like this:
`<% if @user.id && @user.id == @post.id %>`
wrapped around the button.

Reply:
Meets the requirement:
see a helpful error message if they try to delete another user's post